packages/ui/src/components/cell/cell.native.tsx

```diff
@@ -1,108 +1,71 @@
-import { Switch } from 'react-native';
+import { ReactElement, ReactNode, cloneElement } from 'react';
 
-import { ResponsiveValue, useTheme } from '@shopify/restyle';
+import { useTheme } from '@shopify/restyle';
 
-import { Box, ChevronRightIcon, IconProps, Text, Theme, TouchableOpacity } from '../../../native';
-import { TouchableOpacityProps } from '../button/touchable-opacity.native';
+import {
+  Avatar,
+  ChevronRightIcon,
+  Flag,
+  IconProps,
+  ItemLayout,
+  RadioButton,
+  Theme,
+  Switch as UISwitch,
+} from '../../../native';
+import { Pressable, PressableProps } from '../button/pressable.native';
+import { RadioButtonProps } from '../radio-button/radio-button.native';
+import { SwitchProps } from '../switch/switch.native';
 
-type RegularCellVariant = 'active' | 'inactive' | 'critical';
-type SwitchCellVariant = 'switch';
-type CellVariant = RegularCellVariant | SwitchCellVariant;
-
-function getIconColor(
-  variant: CellVariant
-): ResponsiveValue<keyof Theme['colors'], Theme['breakpoints']> {
-  switch (variant) {
-    case 'active':
-    case 'switch':
-      return 'ink.text-primary';
-    case 'inactive':
-      return 'ink.text-subdued';
-    case 'critical':
-      return 'red.action-primary-default';
-  }
+interface CellProps extends PressableProps<Theme> {
+  caption?: string;
+  icon?: ReactElement<IconProps>;
+  title: string;
+  children: NonNullable<ReactNode>;
 }
 
-function getIconBackgroundColor(
-  variant: CellVariant
-): ResponsiveValue<keyof Theme['colors'], Theme['breakpoints']> {
-  switch (variant) {
-    case 'active':
-    case 'switch':
-      return 'ink.background-secondary';
-    case 'inactive':
-      return 'ink.background-secondary';
-    case 'critical':
-      return 'red.background-primary';
-  }
-}
+export function Root({ caption, icon, title, children, ...props }: CellProps) {
+  const theme = useTheme<Theme>();
+  const itemLayout = <ItemLayout actionIcon={children} captionLeft={caption} titleLeft={title} />;
+  const defaultColor: string = theme.colors['ink.text-primary'];
 
-interface SwitchCell {
-  variant: SwitchCellVariant;
-  switchValue: boolean;
-  toggleSwitchValue(): void;
-}
+  const content = icon ? (
+    <Flag
+      img={
+        <Avatar>
+          {cloneElement<IconProps>(icon, {
```

Comment:
I still don't get why we have to do this. We already have a `cloneElement` in `Icon` which does the same thing: 
```
export function Icon({ children, ...props }: IconProps) {
  const child = React.Children.only(children) as React.ReactElement;

  return (
    <>
      {React.cloneElement(child, {
        color: token('colors.ink.action-primary-default'),
        ...props,
      })}
    </>
  );
}

```

Is it because that's setting them to `ink.action-primary-default` and you want them to be `ink.background-secondary` without passing it all the time?

Reply:
Yeah, I'm not sure why it is nec when a default is already set. Tbh, I think the default color being set here is inaccurate? We don't want the default to be `text.primary`? @edgarkhanzadian that is incorrect? Icons by default are `action-primary-default` which is already applied to all icons.